### ISSUE TITLE
Initializing error

### ISSUE THREAD

**Issue by outsidepro-arts** [hxebolax/Tienda-para-NVDA#15]
The error occures on NVDA launching. Due that, addon does not load.
Initializing log:
INFO - external:globalPlugins.TiendaNVDA.GlobalPlugin.tareasDeRed (17:28:59.083) - Thread-4 (8276):
Failed to load libraries required for the store
INFO - external:globalPlugins.TiendaNVDA.GlobalPlugin.tareasDeRed (17:28:59.083) - Thread-4 (8276):
An error occurred in the libraries::

Expecting value: line 1 column 1 (char 0)
ERROR - stderr (17:28:59.086) - Thread-4 (8276):
Traceback (most recent call last):
ERROR - stderr (17:28:59.095) - Thread-4 (8276):
  File "C:\Users\denis\AppData\Roaming\nvda\addons\TiendaNVDA\globalPlugins\TiendaNVDA\__init__.py", line 123, in tareasDeRed
    ajustes.setup()
ERROR - stderr (17:28:59.113) - Thread-4 (8276):
  File "C:\Users\denis\AppData\Roaming\nvda\addons\TiendaNVDA\globalPlugins\TiendaNVDA\ajustes.py", line 84, in setup
    listaAddonsSave = basedatos.libreriaLocal(listaServidores[selectSRV][2]).fileJsonAddon(2)
ERROR - stderr (17:28:59.121) - Thread-4 (8276):
  File "C:\Users\denis\AppData\Roaming\nvda\addons\TiendaNVDA\globalPlugins\TiendaNVDA\basedatos.py", line 189, in fileJsonAddon
    return json.load(fp)
ERROR - stderr (17:28:59.130) - Thread-4 (8276):
  File "json\__init__.pyc", line 296, in load
ERROR - stderr (17:28:59.137) - Thread-4 (8276):
  File "json\__init__.pyc", line 348, in loads
ERROR - stderr (17:28:59.146) - Thread-4 (8276):
  File "json\decoder.pyc", line 337, in decode
ERROR - stderr (17:28:59.154) - Thread-4 (8276):
  File "json\decoder.pyc", line 355, in raw_decode
ERROR - stderr (17:28:59.168) - Thread-4 (8276):
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
INFO - external:globalPlugins.TiendaNVDA.GlobalPlugin.tareasDeRed (17:28:59.247) - Thread-4 (8276):
Add-on launch canceled.

Addon version: 0.8.2

**Comment by hxebolax** [comment]
Hello first of all excuse my English.

Well that bug I hope to fix it in the next version.

I would like to know if you use the TRANSLATE plugin as I have detected that sometimes it corrupts the json and when it loads it drags the JSON library and affects the loading of the Store.

If so I recommend you to clear the TRANSLATE cache or in the NVDA user directory delete the translation-cache folder or with the original key combination if you did not change NVDA+Ctrl+Shift and F twice quickly.

If you do not use TRANSLATE you can also delete the folder in the NVDA StoreNVDA user and restart NVDA. By doing the latter you will lose any settings you had in the store options for the various add-ons such as channel or if you do not want it to check for updates.

I would also like that before you do any of the above if you could try this version of the store that already contemplates this error and try to fix it and let me know if it went well.

This version what it does is to contemplate the error and re-generate the database of add-ons that are compatible on the server.

If this happens as in the previous cases we will have to reconfigure in store options channels that we want for each add-on with if any do not want to get updates.

https://www.dropbox.com/s/316bmnu8fxykkdl/TiendaNVDA-0.8.2.1.nvda-addon?dl=1

Thank you very much and sorry for the inconvenience.


Translated with www.DeepL.com/Translator (free version)

**Comment by outsidepro-arts** [comment]
No I have no any plugin which named as "Translate" installed.
Before all actions you said, I've decided to use the package you published here and it works. I specially have restart NVDA a few times and got no errors. All updates process correctly. all plug-ins lists get aswell. The installation process now in progress and seems works too.
Should I remove this addon you published on the comment above?

**Comment by hxebolax** [comment]
Hi, first of all thank you again for the report and for testing the add-on I shared.

It is not necessary to delete it since it is a minor update that I have not released that includes fixing the error that you communicated to me.

In the next update 0.8.3 or 0.9 depending on will be incorporated the solution that I applied in the add-on that I shared with you.

It will come out as an update since I will not release any version with the numerical nomenclature that has what I shared with you.

Again thank you very much for your report and again apologies for the inconvenience.
